woori-db/src/actors/history.rs

```diff
@@ -83,9 +83,18 @@ fn get_previous_registry(fractions: Vec<&str>) -> Result<Option<DataRegister>, E
         .to_owned();
     let state = &state[..(state.len() - 1)];
 
-    let resp: Result<DataRegister, Error> = match ron::de::from_str(state) {
+    let resp: Result<(DataRegister, HashMap<String, Types>), Error> = match ron::de::from_str(state)
+    {
         Ok(x) => Ok(x),
         Err(_) => Err(Error::FailedToParseRegistry),
     };
-    Ok(Some(resp?))
+    let resp_insert: Result<DataRegister, Error> = match ron::de::from_str(state) {
```

Comment:
On both matches the code could be simplified to use `map_err`, but that is just code simplification/style.

Reply:
ron::error doesnt convert to the correct error for every case, so it is needed to use this kind of solution